test/cbmc/proofs/requestDataBlock_Http/requestDataBlock_Http_harness.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * AWS IoT Over-the-air Update v3.0.0
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file requestDataBlock_Http_harness.c
+ * @brief Implements the proof harness for requestDataBlock_Http function.
+ */
+/* Http interface includes. */
+#include "ota_http_private.h"
+
+/* Stub required for the proof. */
+OtaHttpStatus_t request( uint32_t rangeStart,
+                         uint32_t rangeEnd )
+{
+    OtaHttpStatus_t status;
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+void requestDataBlock_Http_harness()
+{
+    OtaAgentContext_t * pAgentCtx;
+    OtaFileContext_t fileContext;
+    OtaHttpInterface_t http;
+    OtaInterfaces_t interface;
+    OtaErr_t err;
+    OtaAgentContext_t agent;
+
+    /* Allocating memory to the agent context. The agent can never be NULL since it
+     *  is a globally declared variable. */
+    pAgentCtx = &agent;
+
+    /* Initialize the file context field in the Agent context. */
+    pAgentCtx->fileContext = fileContext;
+    http.request = request;
+    interface.http = http;
+
+    /* File size can never be zero and is verified in the validateAndStartJob() function.*/
```

Comment:
It's not that this file size can never be zero in general, it's just that we are checking it before it is passed to this function. So I would change the wording to something like this (make sure to update it for the letter count):
```suggestion
    /* requestDataBlock_Http assumes that the file size is not zero. This is enforced by the validateAndStartJob function, which is always called before the requestDataBlock_Http function. */
```

Reply:
Updated it in 1454bc7